### PR TITLE
[#378] Update Wiki to remove optional features and clarify voting labels

### LINKED ISSUE #378: Update Wiki to clarify our voting process in GitHub Discussions
## Why

Before, we would apply our voting system directly to the issues (by placing 👍  or 👎 reactions). 

As of now, GitHub Discussions has become a better place for discussing and voting (it supports a built-in poll system)

Also, since we now have separate templates, there's no longer any use for "optional" features 

So, we should update our Wiki accordingly...

## Who Benefits?

The Android team, this update will bring more clarity and a better understanding of how issues are managed

## Acceptance Criteria 

Update the Wiki content:

- Remove everything related to "optional features"

- Clarify the voting process on RFCs (GitHub Discussions): 
   -  RFCs should only be created for chores or features, for bugs we can create an issue directly.
   -  In order to reach a conclusion, at least 50% of the team must cast their vote.
   -  To allow sufficient time for everyone to participate, allow a minimum of one week for voting

- Remove all these labels: `status : essential`, `status : optional`,  `status : rejected` & `status : need voting`

  → Because a feature or chore is already approved/rejected once a discussion has been resolved.

- Issues for bugs can be created without any prior voting.

### PR DESCRIPTION
**Pull request by ryan-conway**
closes #378 

## What happened 👀

- Removed all references to optional features
- Updated the content in `How do we vote on an issue?` section
- Created separate sections detailing how to open an issue and when an issue is approved or rejected

## Insight 📝

N/A

## Proof Of Work 📹

github com_nimblehq_android-templates_blob_499e05c841ce21570cf602ba5a3959a3a9eb977d_ github_wiki_Home md

### CODE REVIEW

File: .github/wiki/Home.md
Comment (Tuubz) [inline .github/wiki/Home.md:41]: This will be added by default now: #379, so thinking if we should rewrite this line 💭 

What do you think? 🙏

**Review by Tuubz — COMMENTED** [review]
I left one small comment 🤖

**Review by Tuubz — APPROVED** [review]

File: .github/wiki/Home.md
Comment (lydiasama) [inline .github/wiki/Home.md:41]:
```suggestion
- If we agree with the proposed issue, we must react with 👍. 
```

IMO, our RFCs are siding come with polling to create some changes. So when we would like to collect the RFC poll result, I don't think only 3 votes are enough. 

I have an idea. I think we could have another section about RFCs that require vote results from half of the members and let this section be relevant to the `proposed issue`. What do you think?

File: .github/wiki/Home.md
Comment (lydiasama) [inline .github/wiki/Home.md:49]:
```suggestion
  - If there are at least 3 of the team members react with 👍, then the issue is approved and the `status : approved` label is added.
  - If  If there are at least 3 of the team members react with 👎, then the issue is rejected and the `status : rejected` label is added.
  - If the vote is tied, then the repository maintainer must resolve it.
  
To facilitate a discussion and reach a decision on a solution, our team follows these steps:
  1. Initiate a discussion using the RFC template.
  2. A solution will be considered approved if any of its options receive at least 50% of the team member votes.
```

What do you think?

Comment (ryan-conway) [reply .github/wiki/Home.md:49]: @lydiasama I like the second part but it makes things a little unclear IMO, it's like there are two choices:
1. Open an issue that only needs 3 votes
2. Open an RFC that needs at least 7 (or 50%) votes

Should we specify which one we should open if we want to suggest a change to the template?
Or would issues be for things like chores and bugs while RFC discussions are for features?

CC: @Tuubz @luongvo

Comment (lydiasama) [reply .github/wiki/Home.md:49]: @ryan-conway Good question. Currently, we do both.

Comment (lydiasama) [reply .github/wiki/Home.md:49]: @ryan-conway Maybe it's time to create a RFC discussion for this? 😄

Comment (thiennguyen0196) [reply .github/wiki/Home.md:49]: @lydiasama I dont quite catch your ideas too 🤔 The first part I think we should have at least 50% of vote instead of just 3 votes, so then it will be consistent for all the types of issue 🤔 

Would you mind explaining it a little bit more why we should have 2 approaches when finalizing an RFC discussion 😄 That would be great

Comment (lydiasama) [reply .github/wiki/Home.md:49]: @thiennguyen0196 For example, when someone finds a bug, if we only have the RFC poll with 50% of the member vote, the process to resolve a bug will be quite cumbersome. Instead, if we create simple issues when there are improvement chores of bugs, we can get just 3 approval and start fixing/implementing them.

File: .github/wiki/Home.md
Comment (AVI5HEK) [inline .github/wiki/Home.md:13]:
```suggestion
- It is a Google-recommended component.
```

**Review by AVI5HEK — COMMENTED** [review]
My two cents.

File: .github/wiki/Home.md
Comment (thiennguyen0196) [reply .github/wiki/Home.md:49]: Got it @lydiasama It is clear now 👍 Thank you for your explanation

**Review by thiennguyen0196 — COMMENTED** [review]
Apart from @lydiasama 's concern, rest LGTM 👍 I will approve once it is resolved @ryan-conway

File: .github/wiki/Home.md
Comment (lydiasama) [inline .github/wiki/Home.md:56]: To make it more step-noticeable, I think we could turn it to number list:

```suggestion
1. Create an issue 💭:

- For feature and chores, we must create an RFC with GitHub Discussions.
- For bugs, we must create an issue directly.

2. Vote on an issue 🗳:

- If we agree with one of the poll options, we must vote for that option.
- If we do not agree with any of the options, we must leave a comment explaining why. 
- If there are differing opinions, then the repository maintainer must resolve it.
- To allow sufficient time for everyone to participate, at least one week is allowed for voting.

3. Conclusion for approved/rejected issues 🕐:

- A conclusion is reached if at least half of the team members cast their vote.
- if the majority vote to work on the issue, then the discussion is locked and an issue is created.
- If the majority vote to reject the issue, then the discussion is locked and no issue is created.
- If the vote is tied, then the repository maintainer must resolve it.
```

**Review by lydiasama — COMMENTED** [review]
It might help clarify our process clearer?. 🤔

File: .github/wiki/Home.md
Comment (Tuubz) [inline .github/wiki/Home.md:54]:
```suggestion
- If the majority vote to work on the issue, then the discussion is locked and an issue is created.
```

**Review by Tuubz — APPROVED** [review]
One small comment 🚀

**Review by lydiasama — APPROVED** [review]

**Review by kaungkhantsoe — APPROVED** [review]

**Review by AVI5HEK — APPROVED** [review]

File: .github/wiki/Home.md
Comment (luongvo) [inline .github/wiki/Home.md:41]: Should we lower the tone here to `should` instead of `must`? Does that mean we always have to create RFC, even for the features or chores with obvious details or simple? 
- Say I would like to submit a ticket to upgrade the Kotlin or Gradle versions, should we vote for that? 
- Or when I would like to add more tests (not bringing anything new), will we have to wait for the voting process?

I'm wondering if this will slow down our contribution speed. I'm not trying against the RFC & voting process. However, we always have a process for the team to review & discuss the final approach in any contributions (pull requests).

Comment (ryan-conway) [reply .github/wiki/Home.md:41]: Using `should` instead of `must` would definitely give a friendlier message to readers, but I don't think using that word should be taken as an excuse to open any issues without prior voting process. Otherwise we should add details on how issues get voted on back to the readme and we're back to [this issue](https://github.com/nimblehq/android-templates/pull/409#discussion_r1115206580) again 🥴

Comment (Tuubz) [reply .github/wiki/Home.md:41]: @luongvo Since we already made a vote on https://github.com/nimblehq/android-templates/discussions/417, we should proceed with this for now. 

If you don't agree, I think another RFC should be made to optimize the process better 🙏

Comment (luongvo) [reply .github/wiki/Home.md:41]: @Tuubz @ryan-conway Sorry for the question. However, the RFC mentioned **How `should` we suggest changes to the wiki?**, we did not say anything about a MUST or HAVE TO. So by using that word, I still have concern this process will slow down our contribution effort.

Comment (Tuubz) [reply .github/wiki/Home.md:41]: @luongvo I understand your concern, but how do we decide if an issue is "simple" or "obvious"? 

This can be subjective depending on the developer's experience. 

Or... we make the exception for the repository maintainer to create issues without voting 💭

Comment (luongvo) [reply .github/wiki/Home.md:41]: @Tuubz @ryan-conway IMO, everyone in the chapter, in the company or external contributors, can submit new issues, questions, or RFCs. The repository maintainer decides 
- which submitted issue needs to be clarified & should go through the voting process with an RFC.
- which submitted issue is clear and fit to be selected for the next milestone.

**Review by luongvo — APPROVED** [review]
LGTM now 💪

**Review by chornerman — APPROVED** [review]
LGTM 🚀

**Pull request merged**
